### ISSUE TITLE
batch-size and list-size and input_layer size

### ISSUE THREAD

**Issue by mhajiaghayi** [tensorflow/ranking#51]

**Comment by mhajiaghayi** [comment]
From the tf_ranking_libsvm.py with the default arguments, I can see that the batch-size when you pass it to "get_train_inputs" is set to the FLAGS.list_size, which is 32 by default. 
however, when I look at the input-layer dimension,
<tf.Tensor 'groupwise_dnn_v2/group_score/input_layer/concat:0' shape=(900, 136) dtype=float32>
shouldn't it be (100,136)?
I don't know where '900' is coming from. I would assume 900 to be the batch-size but I don't see anywhere it's set to 900.

**Comment by bendersky** [comment]
Could you clarify what exactly you'd like to know about these sizes?

**Comment by bendersky** [comment]
I recommend checking this [notebook](https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_libsvm.ipynb) for more details. In general, the various sizes used in tf-ranking are:
-list size -- the maximum # of documents that we expect to see per query
-batch size -- number of examples in a single training batch
-input_layer size -- the dimensionality of the input layer
-group size -- in case we score multiple items at a time, the size of the group that is scored jointly.

Hope this helps!

**Comment by bendersky** [comment]
Various usage examples can be found here https://github.com/tensorflow/ranking/tree/master/tensorflow_ranking/examples

**Comment by EricBataller** [comment]
hi @bendersky. thanks for the explanation. It is not self evident to me that batch_size always refers to the examples in a single training batch. In the case of a listwise loss function, where the loss is updated on every query (with the list of examples of that query), rather than on every example. Shouldn't the batch_size refer to the number of queries in a training batch in this case? Could you clarify? Thanks in advance!

**Comment by ramakumar1729** [comment]
It depends on what "examples" mean in the context. 

You're explanation makes senseL: a listwise loss function takes in logits and labels of shape `[batch_size, list_size]`. `batch_size` corresponds to number of queries and `list_size` corresponds to maximum number of documents per query.

**Comment by NicoleXCT** [comment]
@ramakumar1729 still confused about list size and group size, if list size is 8 and group size is 1, after the load_libsvm_data function, the 8 documents with the same 'qid' are formed into one sample, then this sample formed by 8 documents outputs logits with dimension=1?Thanks for explaining!
